extensions/applicationinsights-analytics-js/Tests/Unit/src/AnalyticsPlugin.tests.ts

```diff
@@ -2164,60 +2169,66 @@ export class AnalyticsPluginTests extends AITestClass {
         }
     }
 
-    private waitForException: any = (expectedCount:number, action: string = "", includeInit:boolean = false) => [
-        () => {
+    // waitForExceptionPromise for use with _asyncQueue that returns a promise
+    private waitForExceptionPromise(expectedCount: number, action: string = "", includeInit: boolean = false) {
+        const testContext = this._testContext;
+        const self = this; // Capture 'this' context to use in the polling function
+        
+        return PollingAssert.asyncTaskPollingAssert(function () {
             const message = "polling: " + new Date().toISOString() + " " + action;
```

Comment:
these first actions in the first element of the array should be outside of the asyncTaskPollingAssert